src/LndClient/RPC/Katip.hs

```diff
@@ -174,3 +177,28 @@ closeChannelSync env mConn req =
       case upd of
         Just res -> return $ Right res
         Nothing -> closeChannelRecursive mVar0 $ n - 1
+
+trackPaymentSync ::
+  (KatipContext m, MonadUnliftIO m) =>
+  LndEnv ->
+  TrackPaymentRequest ->
+  m (Either LndError Payment)
+trackPaymentSync env req = do
+  mVar <- newEmptyMVar
+  void $
+    Util.spawnLink $
+      trackPaymentV2
+        (void . tryPutMVar mVar)
+        env
+        req
+  waitTrackResult mVar 10
+  where
+    waitTrackResult _ (0 :: Int) = do
+      $(logTM) (newSev env ErrorS) "Track Payment timeout expired"
+      return $ Left $ LndError "Track Payment timeout expired"
+    waitTrackResult mVar0 n = do
+      sleep $ MicroSecondsDelay 1000000
```

Comment:
Maybe let's use smaller sleep? 1sec is quite a lot. If first message is current status of the payment - it should arrive pretty fast, and we need only it

Reply:
yep, decreased 10 times